patch-core.js

```diff
@@ -11,12 +11,35 @@ const https = require('https');
 const patchMarker = "__agent_base_https_request_patched__";
 if (!https.request[patchMarker]) {
   https.request = (function(request) {
-    return function(_options, cb) {
-      let options;
+    return function(_url, _options, cb) {
+      // Support for both
+      // https.request(options[, callback])
+      // https.request(url[, options][, callback])
+      if (typeof _url !== 'string' && !cb) {
+        cb = _options;
+        _options = _url;
+        _url = undefined;
+      }
+      if (!_url && _options) {
+        _url = url.format(_options);
+      }
+      var options;
       if (typeof _options === 'string') {
         options = url.parse(_options);
       } else {
-        options = Object.assign({}, _options);
+        options = url.parse(_url);
+        if (_options) {
+          if (_options && Object.getOwnPropertySymbols && Object.getOwnPropertySymbols(_options)[0]) {
```

Comment:
This code looks repeated, consider abstracting out into a util function?

Reply:
Sounds good!

Reply:
I did abstract this into a re-usable function! Please check the update!